### ISSUE TITLE
[Task] Soft 'Daily Plan' trigger

### ISSUE THREAD

**Issue by KostyaEver** [ever-co/ever-teams#2679]
There should be a way to detect/know that a member has not worked today yet

- true
- false

**Comment by evereq** [comment]
I think we should have some API for that in Gauzy @GloireMutaliko21

**Comment by KostyaEver** [comment]
@GloireMutaliko21 
The system should know if the user tracked time today 'from 0:00am till time now'. 
whole day range 'from 0:00am to 23:59pm'

if the user did not track, and the user is about to start 'Tracking time' the system displays a popup 'Please create Today's Plan'

the main purpose once a day display popup 'Create Plan' if there is no 'Daily Plan' for today

**Comment by GloireMutaliko21** [comment]
@KostyaEver 
I recently implemented one task : https://github.com/ever-co/ever-teams/issues/2471
Please see this short video and let me know if these two issues are not related : https://github.com/ever-co/ever-teams/pull/2635#issue-2355052994

**Comment by KostyaEver** [comment]
They are absolutely different @GloireMutaliko21 

2471 is an option in settings to enforce popup to provide Plan
2679 is to detect if a member tracked time or not (track event)
new task - is without enforce user to provide Plan, it asks only once a day as reminder only

**Comment by GloireMutaliko21** [comment]
All right @KostyaEver 
Sir @evereq ! I'm not sure we should need some additional API for that as it seems that we only need to check if the "total worked today" is not equals to zero ???

**Comment by KostyaEver** [comment]
@GloireMutaliko21 you are right if 'Total worked today' is 0 then the system can check if there is a 'Daily Plan' or not. 
If not then call the popup 'Please add Today's plan'

Since that was broken, I totally forgot that we could use what already exists :)

**Comment by GloireMutaliko21** [comment]
Great @KostyaEver 
Btw, Yesterday I fixed those issues according to update statistics after tracking time : https://github.com/ever-co/ever-teams/pull/2683#issue-2391163449 :rocket:

**Comment by KostyaEver** [comment]
Hi @GloireMutaliko21 

Please see few flows


1. The system tracks if user worked today or not

2. If there is no Plan Today: 
- the system asks in a popup to Create Plan - just ‘OK’ button (not enforce)
- added task(s), but not estimated, selected task attempt to click ‘Start’ timer
- it asks in the popup to provide an estimate per task and planned work hours
- may skip and work out of plan  (click 'Add later' instead 'Cancel')

3. if has a Plan Today (Not all tasks are estimated)
- Attempt to click ‘Start’ it asks in a popup to provide task estimates and planned work hours https://github.com/ever-co/ever-teams/issues/2603
- may skip and work out of plan  (click 'Add later' instead 'Cancel')

4. if has a Plan Today (All tasks are estimated)
- Attempt to click ‘Start’ it asks in a popup to provide ONLY planned work hours
- may skip and work out of plan   (click 'Add later' instead 'Cancel')

**Comment by GloireMutaliko21** [comment]
@KostyaEver Good flows, thanks. I noted, we already have a setting that enforce the user to have a Plan before start tracking time, https://github.com/ever-co/ever-teams/issues/2471, and the daily plan popup logic https://github.com/ever-co/ever-teams/issues/2601.
Think many of flows are already existed in https://github.com/ever-co/ever-teams/pull/2635#issue-2355052994.

So if I get you well, think we can adjust some logic and show that popup even if the enforce plan setting is 'ON' ??
What I mean : 

**If there is no Plan Today**

1. Enforce Plan setting is 'OFF' (Not require having a plan for time tracking) The 'Start Timer' button is activated
- the system asks in a popup to Create Plan - just ‘OK’ button (not enforce)
- added task(s), but not estimated, attempt to click ‘Start’ timer
- it asks in the popup to provide an estimate per task and planned work hours (Today Plan Creation ???)
- may skip and work out of plan (click 'Add later' instead 'Cancel') : make sense because the "require plan"setting is OFF

2. Enforce Plan setting is 'ON'
- All the steps above
-  may skip and work out of plan (click 'Add later' instead 'Cancel') : DOES NOT make sense because the "require plan"setting is ON

**if has a Plan Today (Not all tasks are estimated)**
- Attempt to click ‘Start’ it asks in a popup to provide task estimates and planned work hours
- may skip and work out of plan (click 'Add later' instead 'Cancel') : Skip and work out of plan ONLY IF "Require plan" setting is OFF because working on a plan that has 0 as planned work hours, not sure that it's good

**if has a Plan Today (All tasks are estimated)**
Here we should also check that "require plan" setting, etc...

**Comment by KostyaEver** [comment]
No @GloireMutaliko21 

Enrorce is the same and there is NO option to avoid popups !!! User always must provide reports/plans etc if it asks otherwise user can not start tracking time


The flows I provided is SOFT - that means when Enforce is OFF, and user can see those popups but can anyways skicp them.

it's clear? @GloireMutaliko21

**Comment by GloireMutaliko21** [comment]
So if I try to summarize, the user can see these popups even if the setting is 'OFF'. But the only difference is that if this setting is OFF and he decides to 'skip' the estimated tasks and hours he plans to work, he can still track. But if the parameter is 'ON', then he can only track if everything has been properly defined.
@KostyaEver

**Comment by KostyaEver** [comment]
Exactly @GloireMutaliko21 

Popups exist always, actually skip popups also can always, just difference is that if Enforse is ON, can not track time without provided Plans / Estimates / Planned work hours. if OFF user can ignore / skip all and track time for different tasks even has plan for today with different set of tasks
